extensions/wikia/PortableInfobox/tests/PortableInfoboxParserTagControllerTest.php

```diff
@@ -51,97 +53,50 @@ public function testEmptyInfobox() {
 			$this->parser->getPreprocessor()->newFrame() )[ 0 ];
 		$output = $this->controller->replaceMarkers( $marker );
 
-		$this->assertEquals( $output, '' );
+		$this->assertEquals( $output, '', 'Should be empty' );
 	}
 
-	public function testThemedInfobox() {
-		$text = '<data><default>test</default></data>';
-		$defaultTheme = 'test';
-
-		$marker = $this->controller->renderInfobox( $text, [ 'theme' => $defaultTheme ], $this->parser,
-			$this->parser->getPreprocessor()->newFrame() )[ 0 ];
-		$output = $this->controller->replaceMarkers( $marker );
-
-		$this->assertTrue( $this->checkClassName(
-			$output,
-			PortableInfoboxParserTagController::INFOBOX_THEME_PREFIX . $defaultTheme
-		) );
-	}
-
-	public function testSourceThemedInfobox() {
-		$text = '<data><default>test</default></data>';
-		$themeVariableName = 'variableName';
-		$themeName = 'variable';
-
-		$marker = $this->controller->renderInfobox( $text, [ 'theme-source' => $themeVariableName ], $this->parser,
-			$this->parser->getPreprocessor()->newCustomFrame( [ $themeVariableName => $themeName ] ) )[ 0 ];
-		$output = $this->controller->replaceMarkers( $marker );
-
-		$this->assertTrue( $this->checkClassName(
-			$output,
-			PortableInfoboxParserTagController::INFOBOX_THEME_PREFIX . $themeName
-		) );
-	}
-
-	public function testEmptySourceDefaultThemedInfobox() {
+	/**
+	 * @dataProvider themeNamesProvider
+	 */
+	public function testThemes( $staticTheme, $variableTheme, $classes, $message ) {
 		$text = '<data><default>test</default></data>';
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "<data><default>test</default></data>" 7 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1192)